pkg/tests/accounts_test.go

```diff
@@ -452,6 +457,219 @@ func Test_ImplicitAccounts(t *testing.T) {
+				testsuite.DefaultSlotDurationInSeconds,
+				8,
+			),
+			iotago.WithLivenessOptions(
+				testsuite.DefaultLivenessThresholdLowerBoundInSeconds,
+				testsuite.DefaultLivenessThresholdUpperBoundInSeconds,
+				testsuite.DefaultMinCommittableAge,
+				100,
+				testsuite.DefaultEpochNearingThreshold,
+			),
+		),
+	)
+	defer ts.Shutdown()
+
+	// Add a validator node to the network. This will add a validator account to the snapshot.
+	node1 := ts.AddValidatorNode("node1")
+	node2 := ts.AddNode("node2")
+
+	validatorBIC := iotago.MaxBlockIssuanceCredits / 2
+	// Add a default block issuer to the network. This will add another block issuer account to the snapshot.
+	walletBIC := iotago.BlockIssuanceCredits(499)
+	wallet := ts.AddGenesisWallet("default", node2, walletBIC)
+
+	ts.Run(false)
+
+	// check that the accounts added in the genesis snapshot were added to the account manager correctly.
+	// validator node account.
+	validatorAccountOutput := ts.AccountOutput("Genesis:1")
+	ts.AssertAccountData(&accounts.AccountData{
+		ID:              node1.Validator.AccountID,
+		Credits:         accounts.NewBlockIssuanceCredits(validatorBIC, 0),
+		OutputID:        validatorAccountOutput.OutputID(),
+		ExpirySlot:      iotago.MaxSlotIndex,
+		BlockIssuerKeys: node1.Validator.BlockIssuerKeys(),
+		StakeEndEpoch:   iotago.MaxEpochIndex,
+		ValidatorStake:  mock.MinValidatorAccountAmount,
+	}, ts.Nodes()...)
+	// default wallet block issuer account.
+	blockIssuerAccountOutput := ts.AccountOutput("Genesis:2")
+	ts.AssertAccountData(&accounts.AccountData{
+		ID:              wallet.BlockIssuer.AccountID,
+		Credits:         accounts.NewBlockIssuanceCredits(walletBIC, 0),
+		OutputID:        blockIssuerAccountOutput.OutputID(),
+		ExpirySlot:      iotago.MaxSlotIndex,
+		BlockIssuerKeys: wallet.BlockIssuer.BlockIssuerKeys(),
+	}, ts.Nodes()...)
+
+	// MODIFY EXISTING GENESIS ACCOUNT
+	var block1Slot iotago.SlotIndex = 1
+	var latestParent *blocks.Block
+	// Issue one block from each of the two block-issuers - one will go negative and the other has enough BICs.
+	{
+		block1Commitment := iotago.NewEmptyCommitment(ts.API.ProtocolParameters().Version())
+		block1Commitment.ReferenceManaCost = ts.API.ProtocolParameters().CongestionControlParameters().MinReferenceManaCost
+		block11 := ts.IssueBasicBlockAtSlotWithOptions("block1.1", block1Slot, ts.Wallet("node1"), &iotago.TaggedData{}, mock.WithSlotCommitment(block1Commitment))
+		block12 := ts.IssueBasicBlockAtSlotWithOptions("block1.2", block1Slot, wallet, &iotago.TaggedData{}, mock.WithStrongParents(block11.ID()), mock.WithSlotCommitment(block1Commitment))
+
+		// Commit BIC burns and check account states.
+		latestParent = ts.CommitUntilSlot(ts.BlockID("block1.2").Slot(), block12)
+
+		burned := iotago.BlockIssuanceCredits(block11.WorkScore()) * iotago.BlockIssuanceCredits(ts.API.ProtocolParameters().CongestionControlParameters().MinReferenceManaCost)
+
+		validatorBIC -= burned
```

Comment:
This would become wallet1BIC and wallet2BIC rather having validatorBIC. There is no reason for this other one to be a validator so its more clear this way.